doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/user-guides/config.md

```diff
@@ -1,9 +1,383 @@
+Kubernetes resource limits specified in the group spec’s pod
+`template`. However, it is sometimes useful to override this autodetected
+value. For example, setting `num-cpus:"0"` for the Ray head pod will prevent Ray
+workloads with non-zero CPU requirements from being scheduled on the head.
+Note that the values of all Ray start parameters, including `num-cpus`,
+must be supplied as **strings**.
+
+#### num-gpus
+This optional field specifies the number of GPUs available to the Ray container.
+In KubeRay versions since 0.3.0, the number of GPUs can be auto-detected from Ray container resource limits.
+For certain advanced use-cases, you may wish to use `num-gpus` to set an {ref}`override<kuberay-gpu-override>`.
+Note that the values of all Ray start parameters, including `num-gpus`,
+must be supplied as **strings**.
+
+#### memory
+The memory available to the Ray is detected automatically from the Kubernetes resource
+limits. If you wish, you may override this autodetected value by setting the desired memory value,
+in bytes, under `rayStartParams.memory`.
+Note that the values of all Ray start parameters, including `memory`,
+must be supplied as **strings**.
+
+#### resources
+This field can be used to specify custom resource capacities for the Ray pod.
+These resource capacities will be advertised to the Ray scheduler and Ray autoscaler.
+For example, the following annotation will mark a Ray pod as having 1 unit of `Custom1` capacity
+and 5 units of `Custom2` capacity.
+```yaml
+rayStartParams:
+    resources: '"{\"Custom1\": 1, \"Custom2\": 5}"'
+```
+You can then annotate tasks and actors with annotations like `@ray.remote(resources={"Custom2": 1})`.
+The Ray scheduler and autoscaler will take appropriate action to schedule such tasks.
+
+Note the format used to express the resources string. In particular, note
+that the backslashes are present as actual characters in the string.
+If you are specifying a RayCluster programmatically, you may have to
+[escape the backslashes](https://github.com/ray-project/ray/blob/cd9cabcadf1607bcda1512d647d382728055e688/python/ray/tests/kuberay/test_autoscaling_e2e.py#L92) to make sure they are processed as part of the string.
+
+The field `rayStartParams.resources` should only be used for custom resources. The keys
+`CPU`, `GPU`, and `memory` are forbidden. If you need to specify overrides for those resource
+fields, use the Ray start parameters `num-cpus`, `num-gpus`, or `memory`.
+
+
+(kuberay-autoscaling-config)=
+## Autoscaler configuration
+```{note}
+If you are deciding whether to use autoscaling for a particular Ray application,
+check out this {ref}`discussion<autoscaler-pro-con>`.
+```
+To enable the optional Ray Autoscaler support, set `enableInTreeAutoscaling:true`.
+The KubeRay operator will then automatically configure an autoscaling sidecar container
+for the Ray head pod. The autoscaler container collects resource metrics from the Ray head container
```

Comment:
better say `collects resource metrics from the Ray cluster` to avoid confusion (or otherwise users might not understand that it takes metrics from the worker containers into considerations too)

Reply:
done